src/ape/types/trace.py

```diff
@@ -219,3 +221,150 @@ class TraceFrame(BaseInterfaceModel):
     """
     The raw trace frame from the provider.
     """
+
+
+class Traceback(list):
```

Comment:
With pydantic, you can do this:

```py
class Traceback(BaseModel):
    __root__: List[TracebackItem]

    ...
```

Reply:
That isn't quite the same! They are still models wrapped around it and not a subclass directly. Example:

```python
from pydantic import BaseModel
from typing import List

class MyList(BaseModel):
    __root__: List[int] = []

class MyStr(BaseModel):
    __root__: str = ""

my_ls = MyList()
my_str = MyStr()
print(f"has append: {hasattr(my_ls, 'append')}")
print(f"str of custom str: {my_str}")
```

Outputs:

```
has append: False
str of custom str: __root__=''
```

This is why in `ethpm-types`, we had to add the `__str__` and `__repr__` methods still when we did `__root__: str`, 'member?

Reply:
I ended up doing this!